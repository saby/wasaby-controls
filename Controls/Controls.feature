<?xml version="1.0" encoding="UTF-8" ?>
<features version="1.00">

  <feature id="menu_mode_selector" last_changed="Золотова Э.Е." last_changed_uuid="55dd1897-4ba4-4c0e-b17e-e05fc0a86623" project="https://online.sbis.ru/opendoc.html?guid=22249abd-c279-4400-a71a-a1b986bf2a0c&amp;client=3" responsible="Золотова Э.Е." responsible_uuid="55dd1897-4ba4-4c0e-b17e-e05fc0a86623">
    <description>Включает отображение меню в виде справочника</description>
  </feature>

  <feature id="new-button-shadow" last_changed="Мочалов М.А." last_changed_uuid="f6fdf353-3ab3-4dbb-8671-01c2b418bf0a" project="https://project.sbis.ru/uuid/58c5e7d5-bbd5-498e-8468-a7495197c9ba/page/project-main" responsible="Мочалов М.А." responsible_uuid="f6fdf353-3ab3-4dbb-8671-01c2b418bf0a">
    <description>Включает новое отображение тени кнопки</description>
  </feature>

</features>
