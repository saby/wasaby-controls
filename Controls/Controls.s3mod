<?xml version="1.0" encoding="UTF-8" ?>
<ui_module id="f6fdf353-3ab3-4dbb-8671-01c2b418bf0a" kaizen_zone="916d4071-6bb4-4800-b3ff-2ee6725c9fdc"  name="Controls" package="Wasaby" responsible="Крайнов Д.О." responsible_uuid="0a464b92-3129-49f2-8a5e-7ce6ecfa37b0" required="1" version="1.00">

  <description>Готовый набор визуальных компонентов, отвечающих корпоративным стандартам. Позволяют быстро спроектировать визуальную составляющую веб-приложения.</description>

    <depends>
      <module id="686f183d-0bcc-4732-b4e4-6016c2a67a0f" name="Types"/>
      <module id="323fd54f-1fab-42a6-8aea-6726d6392d53" name="Application"/>
      <module id="63e1a7e5-4d75-42fa-91a3-977dc90ff1e1" name="SbisEnv"/>
      <module id="0df0fce3-53ba-47f8-987d-edf1fd078295" name="WS.Core"/>
      <module id="636177e5-4d75-44fa-9173-977dc90ff111" name="Env"/>
      <module id="4220f331-8217-437b-810c-8c4479323a27" name="EnvConfig"/>
      <module id="2274d173-fe30-4fcf-81ca-767915919d69" name="EnvTouch"/>
      <module id="656177e2-4d75-42fa-9173-977dc90ff121" name="Browser"/>
      <module id="10efeac7-0f02-4c73-9474-0d1d0db077b6" name="View"/>
      <module id="0f5d26db-dd61-4957-9375-8861ad66f219" name="Vdom"/>
      <module id="f27abe12-19df-4c9f-b886-634bd20a0011" name="UI"/>
      <module id="832cda0a-e19c-4ffc-a1d1-47edbb6b7501" name="I18n"/>
      <module id="37e402f1-2fb0-4b73-81c5-113cf94dbf1c" name="WasabyLoader"/>
      <module id="e4bbd0f2-f7ce-404e-88b2-5130d105b770" name="ParametersWebAPI"/>
      <module id="6ae4f1d6-9ecf-4805-a98c-65e47906eedd" name="Tailwind"/>
      <module id="50f80520-0983-4049-b17a-d893bce91d69" name="ErrorHandling"/>
      <module id="8298f3a2-1964-11ed-861d-0242ac120002" name="Controls-DataEnv"/>
    </depends>

  <load_after/>

  <parameters/>

  <resources/>

  <preload/>

</ui_module>
