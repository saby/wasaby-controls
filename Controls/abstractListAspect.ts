export {
    AbstractAspectStateManager,
    TAbstractAspectStateManagerProps,
} from 'Controls/ListAspects/_abstractListAspect/abstract/AbstractAspectStateManager';
export {
    IListChange,
    ListChangeNameEnum,
    IGetListChangeByName,
    IMoveMarkerListChange,
} from 'Controls/ListAspects/_abstractListAspect/common/ListChanges';

export {
    IStateWithItems,
    copyStateWithItems,
} from 'Controls/ListAspects/_abstractListAspect/common/IStateWithItems';
export {
    copyStateWithHierarchyItems,
    IStateWithHierarchyItems,
} from 'Controls/ListAspects/_abstractListAspect/common/IStateWithHierarchyItems';

export * as CommonUILogic from 'Controls/ListAspects/_abstractListAspect/common/UILogic';
