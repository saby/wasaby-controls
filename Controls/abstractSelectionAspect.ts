export {
    IAbstractSelectionState,
    copyAbstractSelectionState,
    TSelectionModel,
    TSelectionModelStatus,
} from 'Controls/ListAspects/selection/_abstractSelectionAspect/IAbstractSelectionState';
export {
    ISelectionChange,
    SelectionChangeName,
    TSelectionChangeName,
    TSelectionChanges,
} from 'Controls/ListAspects/selection/_abstractSelectionAspect/TSelectionChanges';
export { ISelectionStrategy } from 'Controls/ListAspects/selection/_abstractSelectionAspect/ISelectionStrategy';
export {
    AbstractSelectionStateManager,
    TAbstractSelectionStateManagerProps,
} from 'Controls/ListAspects/selection/_abstractSelectionAspect/AbstractSelectionStateManager';
export * as Utils from 'Controls/ListAspects/selection/_abstractSelectionAspect/Utils';
