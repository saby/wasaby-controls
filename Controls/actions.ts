/**
 * @kaizen_zone ddbc0bdc-0710-4e01-9472-8d1982a63a4e
 */
/**
 * Библиотека стандартных экшенов.
 * @library
 * @public
 * @includes ListAction Controls/_actions/ListActions/ListAction
 * @includes BaseAction Controls/_actions/BaseAction
 * @includes IActionOptions Controls/_actions/interface/IActionOptions
 * @includes IActionProps Controls/_actions/interface/IActionProps
 * @includes ISortActionOptions Controls/_actions/SortingActions/interface/ISortActionOptions
 */

import { IListActionOptions, IListAction } from 'Controls/_actions/interface/IListAction';
import { IActionState, IAction, IActionExecuteParams } from 'Controls/_actions/interface/IAction';
import { IActionOptions } from 'Controls/_actions/interface/IActionOptions';

export { default as BaseAction } from './_actions/BaseAction';
export { default as MassAction } from './_actions/ListActions/ListAction';
export { default as ListAction } from './_actions/ListActions/ListAction';
export { default as Remove } from './_actions/ListActions/Remove';
export { default as ViewMode } from './_actions/ListActions/ViewMode';
export { default as Move } from './_actions/ListActions/Move';

export {
    ISortActionOptions,
    ISortingItem,
} from './_actions/SortingActions/interface/ISortingOptions';
export { default as Sort } from './_actions/SortingActions/Sort';
export { default as SortingMenuItemTemplate } from './_actions/SortingActions/MenuItemTemplate';
export { default as Container, IContainerOptions } from 'Controls/_actions/Container';
export { createAction, executeAction } from 'Controls/_actions/callActionUtils';
export {
    IActionState,
    IListAction,
    IListActionOptions,
    IAction,
    IActionExecuteParams,
    IActionOptions,
};
export { IActionExecuteParams as IExecuteOptions };
