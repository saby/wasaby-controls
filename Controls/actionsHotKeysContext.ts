export { useDispatch } from './_actionsHotKeysContext/hooks/useDispatch';
export { useRegister } from './_actionsHotKeysContext/hooks/useRegister';
export { HotKeysContext } from './_actionsHotKeysContext/Context/HotKeysContext';
export { ContextProvider } from './_actionsHotKeysContext/Context/HotKeysProvider';
