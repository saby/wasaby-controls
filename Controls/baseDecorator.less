@import 'Controls/_baseDecorator/Money/_Money';
@import 'Controls/_baseDecorator/Number/_Number';
@import 'Controls/_baseDecorator/Date/_Date';
@import 'Controls/_baseDecorator/Highlight/_Highlight';
