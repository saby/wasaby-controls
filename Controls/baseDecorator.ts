/**
 * Библиотека базовых декораторов.
 * @library
 * @includes IDate Controls/_baseDecorator/IDate
 * @includes IHighlight Controls/_baseDecorator/IHighlight
 * @includes IMoney Controls/_baseDecorator/IMoney
 * @includes INumber Controls/_baseDecorator/INumber
 * @public
 */

export { default as Money } from './_baseDecorator/Money';
export { default as Number } from './_baseDecorator/Number';
export { default as Date, IDateOptions } from './_baseDecorator/Date';
export {
    default as Highlight,
    IHighlightOptions,
} from './_baseDecorator/Highlight';

export {
    IMoneyOptions,
    calculateFractionFontSize,
} from './_baseDecorator/resources/Money';
export {
    default as IOnlyPositive,
    IOnlyPositiveOptions,
} from './_baseDecorator/interfaces/IOnlyPositive';
export { INumberOptions, RoundMode } from './_baseDecorator/resources/Number';
export { default as toString } from './_baseDecorator/inputUtils/toString';
export * from './_baseDecorator/interfaces/IMask';
export { IFormat } from './_baseDecorator/resources/FormatBuilder';
export {
    phoneMask,
    getAreaCode,
    REPLACER,
    FORMAT_MASK_CHARS,
} from './_baseDecorator/resources/phoneMask';
export * from './_baseDecorator/resources/Util';

export * as NumberFunctions from './_baseDecorator/resources/Number';
export * as MoneyFunctions from './_baseDecorator/resources/Money';
export * as HighlightFunctions from './_baseDecorator/resources/Highlight';

import {
    default as splitIntoTriads,
    concatTriads,
    NUMBER_DIGITS_TRIAD,
    SPLITTER,
} from './_baseDecorator/inputUtils/splitIntoTriads';
import * as Formatter from './_baseDecorator/resources/Formatter';
import * as FormatBuilder from './_baseDecorator/resources/FormatBuilder';
import * as Parser from './_baseDecorator/resources/parse';
import * as numberConstant from './_baseDecorator/resources/NumberConstant';
import { default as numberToString } from './_baseDecorator/inputUtils/numberToString';
import {
    partOfNumber,
    escapeSpecialChars,
    addWordCheck,
} from './_baseDecorator/inputUtils/RegExp';

export {
    NUMBER_DIGITS_TRIAD,
    SPLITTER,
    Formatter,
    FormatBuilder,
    Parser,
    numberConstant,
    partOfNumber,
    numberToString,
    concatTriads,
    splitIntoTriads,
    escapeSpecialChars,
    addWordCheck,
};
