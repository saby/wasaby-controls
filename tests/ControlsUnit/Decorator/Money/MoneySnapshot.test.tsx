/**
 * @jest-environment jsdom
 */
import * as React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { Money } from 'Controls/baseDecorator';

describe('Controls/baseDecorator:Money', function () {
    let container = null;
    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });
    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });
    it('default', () => {
        act(() => {
            render(<Money value="100000" />, container);
        });
        expect(container).toMatchSnapshot();
    });
    it('useGrouping=false', () => {
        act(() => {
            render(<Money value="100000" useGrouping={false} />, container);
        });
        expect(container).toMatchSnapshot();
    });
    it('currency=Dollar', () => {
        act(() => {
            render(<Money value="100000" currency="Dollar" />, container);
        });
        expect(container).toMatchSnapshot();
    });
    it('currencyPosition=left', () => {
        act(() => {
            render(<Money value="100000" currency="Dollar" currencyPosition="left" />, container);
        });
        expect(container).toMatchSnapshot();
    });
    it('currencyStyle', () => {
        [
            'primary',
            'secondary',
            'success',
            'warning',
            'danger',
            'unaccented',
            'link',
            'label',
            'info',
            'default',
            'contrast',
        ].forEach((style) => {
            act(() => {
                render(<Money value="100000" currency="Dollar" currencyStyle={style} />, container);
            });
            expect(container).toMatchSnapshot();
        });
    });
    it('currencySize', () => {
        ['xs', 's', 'm', 'l', 'xl', '2xl', '3xl', '4xl', '5xl', '6xl', '7xl', '8xl'].forEach(
            (size) => {
                act(() => {
                    render(<Money value={1000} currency="Dollar" currencySize={size} />, container);
                });
                expect(container).toMatchSnapshot();
            }
        );
    });
    it('abbreviationType=long', () => {
        [1000, 1000000, 1000000000, 1000000000000].forEach((num) => {
            act(() => {
                render(<Money value={num} abbreviationType="long" />, container);
            });
            expect(container).toMatchSnapshot();
        });
    });
});
