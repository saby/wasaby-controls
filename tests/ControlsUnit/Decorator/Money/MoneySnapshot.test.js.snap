// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/baseDecorator:Money abbreviationType=long 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 тыс
      </span>
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money abbreviationType=long 2`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 млн
      </span>
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money abbreviationType=long 3`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000 000 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 млрд
      </span>
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money abbreviationType=long 4`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000 000 000 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 трлн
      </span>
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currency=Dollar 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyPosition=left 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class="controls-fontsize-s controls-text-default controls-fontweight-default"
    >
      $
    </span>
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default controls-margin_left-2xs"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs"
      >
        .00
      </span>
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-xs controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 2`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 3`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-m controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 4`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-l controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 5`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 6`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-2xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 7`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-3xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 8`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-4xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 9`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-5xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 10`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-6xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 11`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-7xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencySize 12`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="1 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        1 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-8xl controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-primary controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 2`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-secondary controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 3`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-success controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 4`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-warning controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 5`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-danger controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 6`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-unaccented controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 7`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-link controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 8`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-label controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 9`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-info controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 10`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-default controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money currencyStyle 11`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs controls-margin_right-2xs"
      >
        .00
      </span>
    </span>
    <span
      class="controls-fontsize-s controls-text-contrast controls-fontweight-default"
    >
      $
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money default 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100 000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100 000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs"
      >
        .00
      </span>
    </span>
  </span>
</div>
`;

exports[`Controls/baseDecorator:Money useGrouping=false 1`] = `
<div>
  <span
    class="controls-DecoratorMoney controls-text-default"
    title="100000.00"
  >
    <span
      class=""
    >
      <span
        class="controls-fontsize-m controls-text-default controls-fontweight-default"
      >
        100000
      </span>
      <span
        class="controls-DecoratorMoney__fraction__colorStyle-readonly controls-text-readonly controls-fontsize-xs"
      >
        .00
      </span>
    </span>
  </span>
</div>
`;
